bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/cache/ExpiringCache.java

```diff
@@ -28,10 +29,11 @@
  *
  * @param <V> the type of the value
  */
+@NonNullByDefault
 public class ExpiringCache<V> {
     private final long expiry;
     private final Supplier<V> action;
-    private SoftReference<V> value;
+    private SoftReference<V> value = new SoftReference<>(null);
```

Comment:
@martinvw Do I have to add the `@Nullable` annotation to the type parameter too?

Reply:
Yes, because AFAIK that was the problem @Hilbrand was facing because he could not pass a nullable supplier to the Cache